packages/test/service-load-test/src/faultInjectionDriver.ts

```diff
@@ -0,0 +1,173 @@
+        return this._lastDeltaStream;
+    }
+
+    async connectToStorage(): Promise<IDocumentStorageService> {
+        return this.internal.connectToStorage();
+    }
+
+    async connectToDeltaStorage(): Promise<IDocumentDeltaStorageService> {
+        return this.internal.connectToDeltaStorage();
+    }
+
+    async connectToDeltaStream(client: IClient): Promise<IDocumentDeltaConnection> {
+        assert(
+            this._lastDeltaStream?.closed !== false,
+            "Document service factory should only have one open connection");
+        const internal = await this.internal.connectToDeltaStream(client);
+        this._lastDeltaStream = new FaultInjectionDocumentDeltaConnection(internal);
+        return this._lastDeltaStream;
+    }
+
+    /**
+     * Returns the error tracking service
+     */
+    getErrorTrackingService(): IErrorTrackingService | null {
+        return this.internal.getErrorTrackingService();
+    }
+}
+
+export interface FaultInjectionDocumentDeltaConnectionEvents extends IEvent{
+    (event: "sumbit" | "submitSignal",listener: () => void);
+}
+
+export class FaultInjectionDocumentDeltaConnection
+extends EventForwarder<IDocumentDeltaConnectionEvents> implements IDocumentDeltaConnection {
+    private _closed: boolean = false;
+    constructor(private readonly internal: IDocumentDeltaConnection) {
+        super(internal);
+    }
+
+    public get closed() {return this._closed;}
+
+    public get clientId() { return this.internal.clientId;}
+
+    public get claims() { return this.internal.claims;}
+
+    public get mode() { return this.internal.mode;}
+    public get existing() { return this.internal.existing;}
+    public get maxMessageSize() { return this.internal.maxMessageSize;}
+    public get version() { return this.internal.version;}
+    public get initialMessages() { return this.internal.initialMessages;}
+
+    public get initialSignals() { return this.internal.initialSignals;}
+    public get initialClients() { return this.internal.initialClients;}
+    public get serviceConfiguration() { return this.internal.serviceConfiguration;}
+    public get checkpointSequenceNumber() { return this.internal.checkpointSequenceNumber;}
+
+    /**
+     * Submit a new message to the server
+     */
+    submit(messages: IDocumentMessage[]): void {
+        this.internal.submit(messages);
+    }
+
+    /**
+     * Submit a new signal to the server
+     */
+    submitSignal(message: any): void {
+        this.internal.submitSignal(message);
+    }
+
+    /**
+     * Disconnects the given delta connection
+     */
+    close(): void {
+        this._closed = true;
+        this.internal.close();
+    }
+
+    public injectNack(docId: string) {
+        assert(!this.closed, "cannot inject nack into closed delta connection");
+        const nack: Partial<INack> = {
+            content:{
+                code: 500,
+                message: "FaultInjectionNack",
+                type: NackErrorType.BadRequestError,
```

Comment:
Surprised to see `BadRequestError` paired with code `500`

Reply:
yeah. not a lot of options, and it's not even used, just had to pick something